src/main/io/gps.c

```diff
@@ -573,110 +848,260 @@ static void ubloxSetMessageRate(uint8_t messageClass, uint8_t messageID, uint8_t
-    if (gpsConfig()->sbas_integrity) {
-        tx_buffer.payload.cfg_sbas.usage |= UBLOX_USAGE_INTEGRITY;
+        ubloxSendConfigMessage(&tx_buffer, MSG_CFG_VALSET, offsetof(ubxCfgValSet_t, cfgData) + offset, true);
+    } else {
+        //NOTE: default ublox config for sbas mode is: UBLOX_MODE_ENABLED, test is disabled
+        tx_buffer.payload.cfg_sbas.mode = UBLOX_MODE_TEST;
+        if (gpsConfig()->sbasMode != SBAS_NONE) {
+            tx_buffer.payload.cfg_sbas.mode |= UBLOX_MODE_ENABLED;
+        }
+
+        // NOTE: default ublox config for sbas mode is: UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR, integrity is disabled
+        tx_buffer.payload.cfg_sbas.usage = UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR;
+        if (gpsConfig()->sbas_integrity) {
+            tx_buffer.payload.cfg_sbas.usage |= UBLOX_USAGE_INTEGRITY;
+        }
+
+        tx_buffer.payload.cfg_sbas.maxSBAS = 3;
+        tx_buffer.payload.cfg_sbas.scanmode2 = 0;
+        switch (gpsConfig()->sbasMode) {
+            case SBAS_AUTO:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0;
+                break;
+            case SBAS_EGNOS:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0x00010048; //PRN123, PRN126, PRN136
+                break;
+            case SBAS_WAAS:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0x0004A800; //PRN131, PRN133, PRN135, PRN138
+                break;
+            case SBAS_MSAS:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0x00020200; //PRN129, PRN137
+                break;
+            case SBAS_GAGAN:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0x00001180; //PRN127, PRN128, PRN132
+                break;
+            default:
+                tx_buffer.payload.cfg_sbas.scanmode1 = 0;
+                break;
+        }
+        ubloxSendConfigMessage(&tx_buffer, MSG_CFG_SBAS, sizeof(ubxCfgSbas_t), false);
     }
+}
 
-    tx_buffer.payload.cfg_sbas.maxSBAS = 3;
-    tx_buffer.payload.cfg_sbas.scanmode2 = 0;
-    switch (gpsConfig()->sbasMode) {
-        case SBAS_AUTO:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0;
-            break;
-        case SBAS_EGNOS:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0x00010048; //PRN123, PRN126, PRN136
-            break;
-        case SBAS_WAAS:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0x0004A800; //PRN131, PRN133, PRN135, PRN138
-            break;
-        case SBAS_MSAS:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0x00020200; //PRN129, PRN137
-            break;
-        case SBAS_GAGAN:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0x00001180; //PRN127, PRN128, PRN132
-            break;
-        default:
-            tx_buffer.payload.cfg_sbas.scanmode1 = 0;
-            break;
+void setSatInfoMessageRate(uint8_t divisor)
+{
+    // enable satInfoMessage at 1:5 of the nav rate if configurator is connected
+    divisor = (isConfiguratorConnected()) ? 5 : 0;
```

Comment:
Suggest using 1Hz update rate
```suggestion
    divisor = (isConfiguratorConnected()) ? 1 : 0;
```